server/src/main/java/org/elasticsearch/search/query/SingleThreadCollectorManager.java

```diff
@@ -1,39 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0 and the Server Side Public License, v 1; you may not use this file except
- * in compliance with, at your election, the Elastic License 2.0 or the Server
- * Side Public License, v 1.
- */
-
-package org.elasticsearch.search.query;
-
-import org.apache.lucene.search.Collector;
-import org.apache.lucene.search.CollectorManager;
-
-import java.io.IOException;
-import java.util.Collection;
-
-/** {@link CollectorManager} that runs in a non-concurrent search. */
-public class SingleThreadCollectorManager implements CollectorManager<Collector, Void> {
```

Comment:
no more need for single threaded collector managers. The only place where we have that need is when field collapsing is used, for which we assert that we do single slices collection directly in QueryPhaseCollectorManager#forCollapsing